libscript/include/pivot/script/Exceptions.hxx

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include <exception>
+#include <string>
+
+namespace pivot::ecs::script
+{
+
+class MixedIndentException : public std::runtime_error
+{
+public:
+    MixedIndentException(const char *msg, const char *line_, size_t linenb_, const char *info_)
+        : std::runtime_error(msg), line(line_), linenb(linenb_), info(info_)
+    {
+    }
+    /// return the error line number
+    size_t get_line_nb() const { return linenb; }
+    /// return the error raw line
+    const char *get_line() const { return line; }
+    /// return the error specification
+    const char *get_info() const { return info; }
+
+private:
+    size_t linenb;
+    const char *line;
+    const char *info;
```

Comment:
UNSAFE 
if you can't be sure that the input char * would be static, you must not store raw, unmanaged pointer.

Line can be a int
and info should be in the message

You can also store your char * as a std::runtime_error.
```suggestion
    size_t get_line_nb() const { return linenb; }
    /// return the error raw line
    const char *get_line() const { return line.what(); }
    /// return the error specification
    const char *get_info() const { return info.what(); }

private:
    size_t linenb;
    std::runtime_error line;
    std::runtime_error info;
```
It is a weird but valid use case. Still, it would be best if you could limit its usage.